capsules/src/sdcard.rs

```diff
@@ -1444,8 +1444,8 @@ impl<'a, A: hil::time::Alarm<'a>> SDCardClient for SDCardDriver<'a, A> {
     fn read_done(&self, data: &'static mut [u8], len: usize) {
         self.kernel_buf.replace(data);
         self.app.map(|app| {
-            #[allow(unused_mut)]
-            let mut read_len: usize = 0;
+            let mut read_len = 0;
+            let read_len_ref = &mut read_len; // needed to mutate read_len in move closure
```

Comment:
Presumably this has been a bug for years, but the sdcard testing app I wrote ignores the size of the the read done callback: https://github.com/tock/libtock-c/blob/master/libtock/sdcard.c#L55

Was the required `#[allow(unused_mut))]` we we upgraded to rust 2018 a hint that something was wrong with the code?

Reply:
yep, that was a clue